### PR TITLE
Issue #9372: Added example of AST for TokenTypes.ANNOTATION_MEMBER_VA…

### LINKED ISSUE #9372: Add example of AST for TokenTypes.ANNOTATION_MEMBER_VALUE_PAIR
https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ANNOTATION_MEMBER_VALUE_PAIR

needs to be extended to show example of AST.
Source to be updated: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

PLEASE READ for more details: https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-AST-in-TokenTypes

### PR DESCRIPTION
**Pull request by AbhinavRajputEXE**
Fixes #9372 Add example of AST for TokenTypes.ANNOTATION_MEMBER_VALUE_PAIR

### CODE REVIEW

**Comment by AbhinavRajputEXE** [comment]
@romani please review this PR.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -4793,6 +4793,26 @@
      * Its children are the name of the member, the assignment literal
      * and the (compile-time constant conditional expression) value.
      *
+     * <p>For example:</p>
+     * <pre>
+     * &#064;MyAnnotation(someField1 = "Hello")
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     *  |--ANNOTATION
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:4802]: Remove `|--` ,Annotation node should be like root, see examples in other nodes

**Review by romani — CHANGES_REQUESTED** [review]
Items:

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (AbhinavRajputEXE) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: @romani I changed that, please check it now, is it correct?

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 2]
```diff
@@ -4793,6 +4793,24 @@
...
      * and the (compile-time constant conditional expression) value.
      *
+     * <p>For example:</p>
+     * <pre>
+     * &#064;MyAnnotation(someField1 = "Hello")
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     * ANNOTATION
+     *  |--AT (&#064;)
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:4803]: you missing ` -&gt; `

please look at example for ANNOTATION_ARRAY_INIT

**Review by romani — CHANGES_REQUESTED** [review]
items

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 2 above)
Comment (AbhinavRajputEXE) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: @romani , I changed the example this time and added the AST for that, previous example was a little bit confusing 😅, please check this.

**Review by romani — APPROVED** [review]

**Review by romani — CHANGES_REQUESTED** [review]
Squash all commits in one

**Comment by AbhinavRajputEXE** [comment]
@romani I've squashed the commits.

**Review by romani — APPROVED** [review]

**Pull request merged**